### PR TITLE
cincinnati: expose customizable testgraph generator

### PR DESCRIPTION
**Pull request by steveej**
This code is useful for generating Graphs for testing purposes and can
be used in other crates as well. In addition to exposing the existing
functionality a builder with a few possible customizations have been
added.

### CODE REVIEW

File: cincinnati/src/lib.rs [hunk 1]
```diff
@@ -734,47 +729,134 @@ mod tests {
...
+            self.edges = edges;
+            self
+        }
+
+        pub fn with_version_template(mut self, version_template: &str) -> Self {
+            self.version_template = version_template.to_string();
+            self
+        }
+
+        pub fn enable_payload_suffix(mut self, enable: bool) -> Self {
```
Comment (vrutkovs) [inline cincinnati/src/lib.rs:781]: It seems this doesn't get tested yet, right?

**Review by vrutkovs — APPROVED** [review]
LGTM. Not sure if its worth including multiarch bits right now, can be done in follow up PRs

File: cincinnati/src/lib.rs (hunk 1 above)
Comment (steveej) [reply cincinnati/src/lib.rs:781]: Right, i's not used in any tests as of the commit it's being added in.

**Review by steveej — COMMENTED** [review]
> LGTM. Not sure if its worth including multiarch bits right now, can be done in follow up PRs

If you're referring to the suffix functionality, it's not restricted to multi-arch, although that's certainly the first use-case for it. I'm fine introducing these options here as opposed to stripping this part out into the other PR. 

Would you mind turning your LGTM into an actual one so it can be merged? :wink:

**Comment by vrutkovs** [comment]
Okay, I think its safe to get merged now, more commits on this are coming.

/lgtm

**Pull request merged**